### PR TITLE
[7.67.x-blue] RHPAM-3709: upgrade maven dependencies to address CVE-2021-26291

### PR DESCRIPTION
**Pull request by baldimir**
Replaces https://github.com/kiegroup/droolsjbpm-integration/pull/3041

### CODE REVIEW

**Review by jstastny-cz — APPROVED** [review]

**Comment by baldimir** [comment]
jenkins retest this

File: kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/pom.xml [hunk 1]
```diff
@@ -59,6 +59,12 @@
     <dependency>
       <groupId>org.jbpm</groupId>
       <artifactId>jbpm-workitems-rest</artifactId>
+      <exclusions>
```
Comment (gitgabrio) [inline kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/pom.xml:62]: This modification breaks lead to [NoClassFoundError/ClassNotFoundError](https://github.com/kiegroup/jbpm-work-items/pull/298#issuecomment-2014741681)

Comment (baldimir) [reply kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/pom.xml:62]: Ok, thanks, I will take a look. This was done to fix duplicate classes.

Comment (gitgabrio) [reply kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/pom.xml:62]: @baldimir 
TBH - I think it is better to modify "downstream" (i.e., where the failure actually happen) .
I pointed out this modification to explain what causes the issue

**Comment by baldimir** [comment]
jenkins retest this

File: kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/pom.xml (hunk 1 above)
Comment (mareknovotny) [reply kie-spring-boot/kie-spring-boot-autoconfiguration/jbpm-spring-boot-autoconfiguration/pom.xml:62]: i agree with https://github.com/kiegroup/droolsjbpm-integration/pull/3044#discussion_r1540599889 it is always better to declare all dependencies you need and do not rely on pulled transitive ones.

**Review by mareknovotny — APPROVED** [review]

**Comment by mareknovotny** [comment]
there is timed out test failure which seems as flaky one

jenkins retest this please

**Comment by baldimir** [comment]
There is a different test failure, so looks like flaky tests to me. @mareknovotny what do you think please?

**Pull request merged**